docs/blog/2019-03-27-security-for-modern-web-frameworks/index.md

```diff
@@ -0,0 +1,135 @@
+
+## Defining Terms
+
+These are not to be considered universal definitions, just how the author will use them going forward.
+
+- **site** - Any website, web-app, or other content you can browse on the internet in a web browser.
+
+- **server-side site** - Any site that pre-processes code at time of request -- whether using node.js, PHP, Python, or any other method -- before delivering it to the user's browser.
+
+- **client-side site** - Any site (GatsbyJS, NuxtJS, NextJS, Single Page App, etc...) where static files(HTML/CSS/JS) are shipped to the browser for processing -- pre-processing may happen at build time, but does not happen time of request.
+
+- **attack vector** - A resource through which to hack a site. Just because an attack vector exists does not mean your site is vulnerable to that attack, however. A server is an attack vector; but, the server requires a vulnerability for an attacker to exploit (use) that server as an attack vector.
+
+## Shared Security Concerns
+
+No matter how you deploy your site, how little JavaScript you use, whether it is sever-side or client-side: your site has attack vectors.
+
+When it comes to client-side sites, if you are going to get hacked, it is likely going to be at the infrastructure or network level:
+
+- **Infrastructure** - This might be your CDN, code repository, DNS, CI, etc. Thanks to the cloud, most of the work securing this layer is done for you. As long as, you are using two-factor authentication (2FA) and a strong password. You are all using 2FA (when possible) and strong passwords, right!? Good.
+
+- **Network** - This kind of attack happens when an attacker can access to the network being used to access your site. It could be a rouge third-party in your cloud provider, or someone sitting across the coffee shop on the same public WiFi as your user. The exact attack methods vary, but are collectively referred to as a [Man-in-the-Middle](https://en.wikipedia.org/wiki/Man-in-the-middle_attack) attack.
+  The defense here is HTTPS (TLS encryption). Other related technologies such as [Cross-Origin-Resource-Sharing](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS), [Strict-Transport-Security](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security), [Public Key Pinning](*https://developer.mozilla.org/en-US/docs/Web/HTTP/Public_Key_Pinning), and [Content Security Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) all assist in securing network traffic. No matter who is looking at your network traffic, they can not read or modify it.
+
+Other possible attacks relate to using `iframes` or not removing referrer tags on external links. Thankfully, fixing these is a pretty low bar. This [article](https://blog.sqreen.com/static-websites-security/) covers all these security issues well and includes solutions and deeper discussion.
```

Comment:
We should name the link text as close to the title/link

```suggestion
Other possible attacks relate to using `iframes` or not removing referrer tags on external links. Thankfully, fixing these is a pretty low bar. The article [Security for Static Websites](https://blog.sqreen.com/static-websites-security/) covers all these security issues well and includes solutions and deeper discussion.
```

Reply:
Good suggestion!

Reply:
I agree, that would be a good change to make. Not seeing it here yet!